tests/modules/programs/gradle/example-settings.nix

```diff
@@ -0,0 +1,33 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  config = {
+    programs.gradle = {
+      enable = true;
+      settings = {
+        "org.gradle.caching" = true;
+        "org.gradle.parallel" = true;
+        "org.gradle.java.home" = pkgs.jdk17;
```

Comment:
Unfortunately you'll need to stub these jdk packages (or just use some other mocked packages), otherwise the test downloads will be enormous just to insert a short text into the output file.

Reply:
Had a look at this and unfortunately it is a bit tricky to accomplish this in this case. For now, I think the best possible solution is

``` patch
diff --git a/tests/modules/programs/gradle/example-settings.nix b/tests/modules/programs/gradle/example-settings.nix
index 36b9736f..19b8c070 100644
--- a/tests/modules/programs/gradle/example-settings.nix
+++ b/tests/modules/programs/gradle/example-settings.nix
@@ -14,17 +14,26 @@ with lib;
       };
     };
 
-    test.stubs.gradle = { };
+    programs.java.package =
+      pkgs.runCommandLocal "java" { home = ""; } "mkdir $out";
+
+    test.stubs = {
+      gradle = { };
+      jdk = { };
+      jdk8 = { };
+      jdk11 = { };
+      jdk17 = { };
+    };
 
     nmt.script = ''
       assertFileExists home-files/.gradle/gradle.properties
       assertFileContent home-files/.gradle/gradle.properties ${
-        pkgs.writeText "gradle.expected" ''
+        builtins.toFile "gradle.expected" ''
           # Generated with Nix
 
           org.gradle.caching = true
-          org.gradle.java.home = ${pkgs.jdk17}
-          org.gradle.java.installations.paths = ${pkgs.jdk8},${pkgs.jdk11}
+          org.gradle.java.home = @jdk17@
+          org.gradle.java.installations.paths = @jdk8@,@jdk11@
           org.gradle.parallel = true
         ''
       }
```

Reply:
Done!